CLUE_Rock_Paper_Scissors/advanced/clue-multi-rpsgame.py

```diff
@@ -0,0 +1,556 @@
+from rps_display import RPSDisplay, blankScreen
+
+
+# Look for our name in secrets.py file if present
+ble_name = None
+try:
+    from secrets import secrets
+    ble_name = secrets.get("rps_name")
+    if ble_name is None:
+        ble_name = secrets.get("ble_name")
+        if ble_name is None:
+            print("INFO: No rps_name or ble_name entry found in secrets dict")
+except ImportError:
+    pass
+
+
+debug = 1
+
+def d_print(level, *args, **kwargs):
+    """A simple conditional print for debugging based on global debug level."""
+    if not isinstance(level, int):
+        print(level, *args, **kwargs)
+    elif debug >= level:
+        print(*args, **kwargs)
+
+
+def tftGizmoPresent():
+    """Determine if the TFT Gizmo is attached.
+       The TFT's Gizmo circuitry for backlight features a 10k pull-down resistor.
+       This attempts to verify the presence of the pull-down to determine
+       if TFT Gizmo is present.
+       This is likely to get confused if anything else is connected to pad A3.
+       Only use this on Circuit Playground Express (CPX)
+       or Circuit Playground Bluefruit (CPB) boards."""
+    present = True
+    try:
+        with digitalio.DigitalInOut(board.A3) as backlight_pin:
+            backlight_pin.pull = digitalio.Pull.UP
+            present = not backlight_pin.value
+    except ValueError:
+        # The Gizmo is already initialised, i.e. showing console output
+        pass
+
+    return present
+
+
+# Assuming CLUE if it's not a Circuit Playround (Bluefruit)
+clue_less = "Circuit Playground" in os.uname().machine
+
+# Note: difference in pull-up and pull-down
+#       and logical not use for buttons
+if clue_less:
+    # CPB with TFT Gizmo (240x240)
+    # from adafruit_circuitplayground import cp  # Avoiding to save memory
+
+    # Outputs
+    if tftGizmoPresent():
+        from adafruit_gizmo import tft_gizmo
+        display = tft_gizmo.TFT_Gizmo()
+        JG_RX_COL = 0x0000ff
+        BUTTON_Y_POS = 120
+    else:
+        display = None
+        JG_RX_COL = 0x000030  # dimmer blue for upward facing CPB NeoPixels
+        BUTTON_Y_POS = None
+
+    audio_out = AudioOut(board.SPEAKER)
+    #pixels = cp.pixels
+    pixels = neopixel.NeoPixel(board.NEOPIXEL, 10)
+
+    # Enable the onboard amplifier for speaker
+    #cp._speaker_enable.value = True  # pylint: disable=protected-access
+    speaker_enable = digitalio.DigitalInOut(board.SPEAKER_ENABLE)
+    speaker_enable.switch_to_output(value=False)
+    speaker_enable.value = True
+
+    # Inputs
+    # buttons reversed if it is used upside-down with Gizmo
+    _button_a = digitalio.DigitalInOut(board.BUTTON_A)
+    _button_a.switch_to_input(pull=digitalio.Pull.DOWN)
+    _button_b = digitalio.DigitalInOut(board.BUTTON_B)
+    _button_b.switch_to_input(pull=digitalio.Pull.DOWN)
+    if display is None:
+        button_left = lambda: _button_a.value
+        button_right = lambda: _button_b.value
```

Comment:
Why lambdas instead of single-line def's?

Reply:
No particular reason. The **pylint** police will be after me if I literally do a one-line `def`. Should I change these to

```
def button_left():
    return _button_a.value
```

then?

Reply:
Do you want to me to change these to `def`s?

Reply:
If you want to explain the code, it's probably easier not to have to explain lambdas.

Reply:
There's a few more. I use them for adapting the args on some callback functions too further down. And one lurks in `rps_display.py` for `key` in `sort()` use for NeoPixel displaying.

Quick hacky survey shows Learn code does use these although 4 of those including this one are mine.

```
$ for fred in *; do echo ${fred} $(cat $(find ${fred} -type f -name '*.py' -print) /dev/null | fgrep -c lambda:); done | grep -v ' 0$'
CircuitPython_JEplayer_mp3 4
CLUE_Dice_Roller 2
CLUE_Metal_Detector 5
CLUE_Rock_Paper_Scissors 18
CLUE_Sensor_Plotter 5
CPB_Quick_Draw_Duo 2
Disc_Step_Sequencer 3
TFT_Gizmo_Turtle_Graphics 7
ulab_Crunch_Numbers_Fast 4
```

I have a slight preference to leave as is but I'm ok with stripping them out too.

Reply:
Lambdas are useful when you don't want to bother to name a function. In this case, it's just a way of circumventing pylint complaints, and I'd prefer using just `def`'s. The black formatter adds a lot of other extra lines too. We run all library code through black; you might consider the same thing, though you might not like the result.

Reply:
Right, I just marked the first instance of these; there are others.